src/cli/cmd/mod.rs

```diff
@@ -367,6 +382,99 @@ macro_rules! flakehub_url {
     }};
 }
 
+fn is_root_user() -> bool {
+    nix::unistd::getuid().is_root()
+}
+
+async fn nix_command(args: &[&str], sudo_if_necessary: bool) -> Result<(), FhError> {
+    command_exists("nix")?;
+
+    let use_sudo = sudo_if_necessary && !is_root_user();
+
+    let mut cmd = if use_sudo {
+        tracing::warn!(
+            "Current user is {} rather than root; running Nix command using sudo",
+            whoami::username()
+        );
+        tracing::debug!(
+            "Running: sudo nix --extra-experimental-features 'nix-command-flakes' {}",
+            args.join(" ")
+        );
```

Comment:
IMO, we should use the debug/display (I forget which it has) implementation of `std::process::Command` right before we spawn it. That way it will always contain every argument we pass.

(Tokio has a helper `.as_std()` that you can use to get the display/debug impl of the std type)